meson.build

```diff
@@ -535,20 +535,39 @@ if thread_local_storage
 endif
 
 if sysroot_install
+  # Get 'sysroot' or 'GCC_EXEC_PREFIX' from GCC output
   sysroot = run_command(cc.cmd_array() + ['-print-sysroot'], check : true).stdout().split('\n')[0]
   if sysroot != ''
-    specs_prefix_format = '%R/@0@'
+    specs_prefix_format_format = '%R/@0@'
+    specs_prefix_format_default = '%R/@0@'
   else
-    sysroot = cc_install_dir + '../../../..'
-    specs_prefix_format = '%:getenv(GCC_EXEC_PREFIX ../../@0@)'
-  endif
-  if not get_option('sysroot-install-skip-checks')
-    if sysroot == ''
+    if not get_option('sysroot-install-skip-checks')
       error('sysroot install requested, but compiler has no sysroot')
     endif
-    if not fs.is_samepath(sysroot, prefix)
-      error('sysroot install requires --prefix=' + sysroot)
+    sysroot = run_command(cc.cmd_array() + ['-print-search-dirs'], check : true).stdout().split('\n')[0].split(' ')[1]
+    specs_prefix_format_format = '%:getenv(GCC_EXEC_PREFIX @0@)'
+    specs_prefix_format_default = '%:getenv(GCC_EXEC_PREFIX ../../@0@)'
+  endif
+
+  # Try to calculate relative path from sysroot to prefix
+  specs_prefix_format = ''
+  if fs.exists(sysroot)
+    sysroot_to_prefix_correction = ''
+    foreach _ : sysroot.split('/')
```

Comment:
Yeah, this looks good. You'll need to add a test after the loop is done to make sure it worked and bail otherwise. With this, it looks like you're just automatically generating the correct path adjustment instead of assuming it is ../../../.. ? If so, I think I'm happy to just merge this if it works for you; it shouldn't be any worse than the current code. Thanks for finding a way to keep it all inside meson instead of running a POSIX command.